src/test/java/com/trainerworkloadservice/service/TrainerInfoMongoServiceTest.java

```diff
@@ -0,0 +1,299 @@
+package com.trainerworkloadservice.service;
+
+import com.trainerworkloadservice.dto.TrainerWorkloadServiceDto;
+import com.trainerworkloadservice.mapper.TrainerInfoMapper;
+import com.trainerworkloadservice.model.Month;
+import com.trainerworkloadservice.model.TrainerInfo;
+import com.trainerworkloadservice.model.Year;
+import com.trainerworkloadservice.repository.TrainerInfoMongoRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trainerworkloadservice.utils.TrainerStatus;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.apache.activemq.command.ActiveMQTextMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class TrainerInfoMongoServiceTest {
+
+  @Mock
+  private TrainerInfoMongoRepository repository;
+
+  @Mock
+  private ObjectMapper objectMapper;
+
+  @Mock
+  private TrainerInfoMapper trainerInfoMapper;
+
+  private ActiveMQTextMessage activeMQTextMessage;
+
+
+
+  @InjectMocks
+  private TrainerInfoMongoService trainerInfoMongoService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    activeMQTextMessage = new ActiveMQTextMessage();
+  }
+
+  @Test
+  void findByUsername_shouldReturnTrainerInfo_whenTrainerExists() {
+
+    String username = "test_trainer";
+    TrainerInfo mockTrainer = new TrainerInfo();
+    mockTrainer.setUsername(username);
+    mockTrainer.setFirstName("John");
+    mockTrainer.setLastName("Doe");
+
+    when(repository.findByUsername(username)).thenReturn(mockTrainer);
```

Comment:
It's okay, but small improvement. try to use any() when you write 'when()' section and add 'verify()' for specific value.